src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,162 @@
-
+        if (index > sizeList || index < 0) {
+            throw new IndexOutOfBoundsException("Incorrect index");
+        }
+        if (index == sizeList) {
+            add(value);
+        } else {
+            linkBefore(value, nodeIndex(index));
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int i = 0; i < list.size(); i++) {
+            T value = list.get(i);
+            add(value);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        isIndexExist(index);
+        return nodeIndex(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        isIndexExist(index);
+        Node<T> currentIndex = nodeIndex(index);
+        T oldVal = currentIndex.element;
+        currentIndex.element = value;
+        return oldVal;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        isIndexExist(index);
+        return position(nodeIndex(index));
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> newNode = head; newNode != null; newNode = newNode.last) {
+                if (newNode.element == null) {
+                    position(newNode);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> nodeNew = head; nodeNew != null; nodeNew = nodeNew.last) {
+                if (object.equals(nodeNew.element)) {
+                    position(nodeNew);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return sizeList;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return sizeList == 0;
+    }
+
+    private void isIndexExist(int index) {
+        if (index < 0 || index >= sizeList) {
+            throw new IndexOutOfBoundsException("Element can't be found!");
+        }
+    }
+
+    Node<T> nodeIndex(int index) {
+        Node<T> newNodeHead = head;
+        Node<T> newNodeTail = tail;
+        if (index < (sizeList >> 1)) {
+            for (int i = 0; i < index; i++) {
+                newNodeHead = newNodeHead.last;
+            }
+        } else {
+            for (int i = sizeList - 1; i > index; i--) {
+                newNodeTail = newNodeTail.first;
+            }
+        }
+        return index < (sizeList >> 1) ? newNodeHead : newNodeTail;
+    }
+
+    void linkBefore(T item, Node<T> current) {
+        Node<T> prev = current.first;
+        Node<T> newNode = new Node<>(prev, item, current);
+        current.first = newNode;
+        if (prev == null) {
+            head = newNode;
+            sizeList++;
+        } else {
+            prev.last = newNode;
+            sizeList++;
+        }
+    }
+
+    T position(Node<T> currentIndex) {
```

Comment:
Let's try to improve naming: 
is input param index or it's a node? 
make method name a verb